checker/checker.go

```diff
@@ -393,7 +393,7 @@ func (v *visitor) MemberNode(node *ast.MemberNode) (reflect.Type, info) {
 		return anyType, info{}
 
 	case reflect.Map:
-		if !prop.AssignableTo(base.Key()) {
+		if v.config.Strict && !prop.AssignableTo(base.Key()) {
```

Comment:
Should the check be `!isAny(prop)`? As from the test example **Key** is **any**.